patroni/log.py

```diff
@@ -4,63 +4,144 @@
 from copy import deepcopy
 from logging.handlers import RotatingFileHandler
 from patroni.utils import deep_compare
+from six.moves.queue import Queue
+from threading import Lock, Thread
 
+_LOGGER = logging.getLogger(__name__)
 
-class PatroniLogger(object):
+
+class QueueHandler(logging.Handler):
+
+    def __init__(self):
+        logging.Handler.__init__(self)
+        self.queue = Queue()
+        self._records_lost = 0
+
+    def _put_record(self, record):
+        self.format(record)
+        record.msg = record.message
+        record.args = None
+        record.exc_info = None
+        self.queue.put_nowait(record)
+
+    def _try_to_report_lost_records(self):
+        if self._records_lost:
+            try:
+                record = _LOGGER.makeRecord(_LOGGER.name, logging.WARNING, __file__, 0,
+                                            'QueueHandler has lost %s log records',
```

Comment:
imo "logging queue" instead of `QueueHandler` is more understandable. Or do you want to explicitly point to the implementation ?